File: app/src/main/java/com/compuspulse/data/remote/model/ApiResponse.java
package com.compuspulse.data.remote.model;

public class ApiResponse<T> {
    private T data;
    private int errorCode;
    private String errorMsg;

    public T getData() {
        return data;
    }

    public int getErrorCode() {
        return errorCode;
    }

    public String getErrorMsg() {
        return errorMsg;
    }

    public boolean isSuccess() {
        return errorCode == 0;
    }
}

File: app/src/main/java/com/compuspulse/data/remote/model/ApiResponse.kt
package com.compuspulse.data.remote.model

class ApiResponse<T> {
    private var data: T? = null
    private var errorCode: Int = 0
    private var errorMsg: String? = null

    fun getData(): T? = data

    fun getErrorCode(): Int = errorCode

    fun getErrorMsg(): String? = errorMsg

    fun isSuccess(): Boolean = errorCode == 0
}
